src/cmds/qstat.c

```diff
@@ -399,18 +401,47 @@ prt_attr(char *n, char *r, char *v, int one_line)
 				printf("%s.%s=", n, r);
 			else
 				printf("%s=", n);
-			escape_delimiter(v, delimiter, '\\', buf);
+			buf = escape_delimiter(v, delimiter, '\\');
+			if(buf == NULL) {
+				fprintf(stderr, "qstat: out of memory\n");
+				exit(1);
+			}
 			printf("%s", buf);
+			free(buf);
 			break;
 
 		default:
 			if(one_line) {
-				replace(v, "\n", "\\n", buf);
-				replace(buf, "\f", "\\f", buf);
+				i = 0; l = MAXPATHLEN;
+				buf = (char *)malloc(MAXPATHLEN);
+				if(buf == NULL) {
+					fprintf(stderr, "qstat: out of memory\n");
+					exit(1);
+				}
+				while(*v != '\0') {
+					if(*v == '\n') {
+						buf[i++] = '\\'; buf[i++] = 'n'; v++;
+					}
+					else if(*v == '\f') {
+						buf[i++] = '\\'; buf[i++] = 'f'; v++;
+					}
+					else
+						buf[i++] = *v++;
+					if(i >= l-2) {
+						l *= 2;
+						buf = (char *)realloc(buf, l);
```

Comment:
if you are using realloc, please store the return value in a seperate variable and use it to free the memory before calling exit() below.